### DISCUSSION TITLE
can support path access please

### DISCUSSION THREAD

**Discussion by liuaifu** [nlohmann/json#3402] [Q&A] [answered]
I'm using boost::property_ptree::ptree. It can set and get value by path, very convenience. But It convert integer value to string, so I'm finding a new json library.

example:

    boost::property_ptree::ptree ptree;
    ptree.put("node1.node2.id", 100);
    auto id = ptree.get<int>("node1.node2.id");

**Comment by nlohmann** [comment]
In order to asses your issue, we need the following information:

- What is the issue you have?

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

- What is the expected behavior?

- And what is the actual behavior instead?

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?

- Did you use a released version of the library or the version from the `develop` branch?

- If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

**Comment by liuaifu** [comment]
Note that this reply is machine translated.
This is not a bug, just a requirements discussion. So the information you requested is not available.
The library is already good, and it would be nice if it could read value by a path like boost::property_ptree::ptree. I'm sure those familiar with Boost will understand the sample code I've provided above.

**Comment by gregmarr** [comment, marked as answer]
I think you're looking for JSON Pointer.

https://github.com/nlohmann/json#json-pointer-and-json-patch
